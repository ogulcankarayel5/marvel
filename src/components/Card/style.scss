.card {
  $root: &;
  height: 30rem;
  position: relative;
  border-radius: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;

  &__img {
    object-fit: cover;
    #{$root}:hover & {
      filter: blur(0.5rem);
      opacity:0.6;
    }
  }

  &__overlay {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;

    visibility: hidden;
    opacity: 0;
    
    /* transition effect. not necessary */
    transition: opacity 0.3s, visibility 0.3s;

    #{$root}:hover & {
      visibility: visible;
      opacity: 1;
    }
  }

  &__title{
      text-align: center;
      color:black;
      font-size: 3rem;
  }

  &__button{
    color:white;
    display:flex;
    justify-content: center;
    align-items: center;
    width:8rem;
    padding:1rem;
    background-color: red;
  }
}
