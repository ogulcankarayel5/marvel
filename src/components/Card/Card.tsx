import React from 'react';
import './style.scss';

function Card({children}:any) {
    return (
         
        <div className="card" >
            {children}
      </div>
    )
}


export default Card

export const CardImage = ({path}:any) => {
   
    return (
        <img data-testid="image" className="card__img" src={`${path}/detail.jpg`} alt="Marvel"/>
    )
}

export const CardOverlay = ({children}:any) => {
    return (
        <div data-testid="overlay" className="card__overlay">
            {children}
        </div>
    )
}
export const CardTitle = ({name}:any) => {
    return (
        <p className="card__title">
            {name}
        </p>
    )
}

export const CardButton = (click:any) => {
    return (
        <button onClick={click} className="card__button">
            More...
        </button>
    )
}