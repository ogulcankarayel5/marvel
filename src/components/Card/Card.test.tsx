import { render } from "@testing-library/react";
import React from "react";
import Card, { CardImage, CardOverlay, CardTitle } from "./Card";

describe("<Card/>", () => {
  it("renders the <Card/> with populated data", () => {
    const { container, getByText,getByTestId } = render(
      <Card>
        <CardImage />
        <CardOverlay>
          <CardTitle name="Title" />
        </CardOverlay>
      </Card>
    );

    expect(getByText('Title')).toBeTruthy();
    expect(getByTestId("image")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();

    
  });

  it("renders the <Card/> without CardOverlay and CardTitle",() => {
      const {container,getByTestId,queryByText,queryByTestId} = render(
          <Card>
              <CardImage/>
          </Card>
      )
      expect(queryByText('Title')).toBeFalsy();
      expect(queryByTestId("overlay")).toBeFalsy();
      expect(getByTestId("image")).toBeTruthy();
      expect(container.firstChild).toMatchSnapshot();
  })

});
