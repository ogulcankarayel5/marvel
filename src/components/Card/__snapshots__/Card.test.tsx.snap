// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders the <Card/> with populated data 1`] = `
<div
  class="card"
>
  <img
    alt="Marvel"
    class="card__img"
    data-testid="image"
    src="undefined/detail.jpg"
  />
  <div
    class="card__overlay"
    data-testid="overlay"
  >
    <p
      class="card__title"
    >
      Title
    </p>
  </div>
</div>
`;

exports[`<Card/> renders the <Card/> without CardOverlay and CardTitle 1`] = `
<div
  class="card"
>
  <img
    alt="Marvel"
    class="card__img"
    data-testid="image"
    src="undefined/detail.jpg"
  />
</div>
`;
